trackers/core/reid/model.py

```
@@ -0,0 +1,477 @@
+            "learning_rate": learning_rate,
+            "optimizer_class": optimizer_class,
+            "optimizer_kwargs": optimizer_kwargs,
+            "random_state": random_state,
+            "projection_dimension": projection_dimension,
+            "freeze_backbone": freeze_backbone,
+            "model_metadata": self.model_metadata,
+        }
+
+        # Initialize callbacks
+        callbacks: list[BaseCallback] = []
+        if log_to_matplotlib:
+            try:
+                from trackers.core.reid.callbacks import MatplotlibCallback
+
+                callbacks.append(MatplotlibCallback())
+            except (ImportError, AttributeError) as e:
+                logger.error(
+                    "Metric logging dependencies are not installed. "
+                    "Please install it using `pip install trackers[metrics]`.",
+                )
+                raise e
+        if log_to_tensorboard:
+            try:
+                from trackers.core.reid.callbacks import TensorboardCallback
+
+                callbacks.append(
+                    TensorboardCallback(
+                        log_dir=os.path.join(log_dir, "tensorboard_logs")
+                    )
+                )
+            except (ImportError, AttributeError) as e:
+                logger.error(
+                    "Metric logging dependencies are not installed. "
+                    "Please install it using `pip install trackers[metrics]`."
+                )
+                raise e
+
+        if log_to_wandb:
+            try:
+                from trackers.core.reid.callbacks import WandbCallback
+
+                callbacks.append(WandbCallback(config=config))
+            except (ImportError, AttributeError) as e:
+                logger.error(
+                    "Metric logging dependencies are not installed. "
+                    "Please install it using `pip install trackers[metrics]`."
+                )
+                raise e
```

should we halt the entire training if the logger fails to load, or just display a warning instead?

---

Imo we should just give warning with extra message like no logging and continue to train

---

IMO, when someone passes `log_to_tensorboard=True`, they're expecting the metrics to be logged to Tensorboard while training, otherwise they would have passed `False`. We should not silently skip logging since its explicitly specified by the user.

---

As long as the `log_to_x` is not set to True as default, I agree with @soumik12345 to `raise` because it will be explicitly set by the user.

As a user, if I set True and I don't see the `wandb` url, I will be confused and in the stdout might miss to see the warning log.

---

However, if it has to fail, it should fail as soon as possible. As a user, I would not like to see error after waiting 2 mins for model loading and 3 minutes for data prep (throwing numbers but you get the point. :)).

---

Let’s set `log_to_tensorboard` and `log_to_wandb` to `False` by default, so the user has to explicitly enable them. And as @ayulockin mentioned, let’s fail fast—raise an error early if logging is enabled but the required dependencies aren’t available.

---

They're set to `False` by default.